making clearing method clear the history element

yaaHosein opened the pull request:
Let us work together!!

yaaHosein committed:
making clearing method clear the history element

SMH110 commented:
I open the files changed tab and this what I saw
![image](https://user-images.githubusercontent.com/16171515/119311877-868e6900-bc69-11eb-8e14-53e877ebbc30.png)


Can you please tell me why you have this much of code being edited while you just added one method?

yaaHosein commented:
Very strange
I did not touched these lines last night
I just added maby only one line inside clearing method and uncomment this method

yaaHosein commented:
> I open the files changed tab and this what I saw
> ![image](https://user-images.githubusercontent.com/16171515/119311877-868e6900-bc69-11eb-8e14-53e877ebbc30.png)
> 
> Can you please tell me why you have this much of code being edited while you just added one method?

Very strange
I did not touched these lines last night
I just added only one or 2 line inside clearing method and uncomment this method

yaaHosein commented:
> I open the files changed tab and this what I saw
> ![image](https://user-images.githubusercontent.com/16171515/119311877-868e6900-bc69-11eb-8e14-53e877ebbc30.png)
> 
> Can you please tell me why you have this much of code being edited while you just added one method?

Very strange
I did not touched these lines last night
I just added maby only one line inside clearing method and uncomment this method

SMH110 commented:
Can you please tell exactly what you did after completing the PR ?

yaaHosein commented:
> Can you please tell exactly what you did after completing the PR ?

As you are aware I had added clearing method a long time ago
So just  uncommented it
And added 2 line
Assigning history to ""
And updating the history

I did the changes and checking first in a different file (draft file( 
I do not remember if by mistake touched anything in the orginal file

Maybe these changes are very old changes
But I do not know if they are old why it dosent pushed eralier

SMH110 commented:
> > Can you please tell exactly what you did after completing the PR ?
> 
> As you are aware I had added clearing method a long time ago
> So just uncommented it
> And added 2 line
> Assigning history to ""
> And updating the history
> 
> I did the changes and checking first in a different file (draft file(
> I do not remember if by mistake touched anything in the orginal file
> 
> Maybe these changes are very old changes
> But I do not know if they are old why it dosent pushed eralier

I mean how did you create the new branch 
did you pull your changes from github ..etc

yaaHosein commented:
Yes I did
First checkout to master
Then git pull
Then creating the new branch and pushing it
Then pushing new changes 
Then pull request

SMH110 commented:
You have removed some code which exist in master 
![image](https://user-images.githubusercontent.com/16171515/119377576-33420800-bcb5-11eb-943d-3440c91df194.png)

I have opened your master branch (M) next to the feature branch (F)
As part of your changes you removed some code was on the master branch.

SMH110 commented on DevelopedCalculator.js:
```diff
@@ -19,17 +19,19 @@ class DevelopedCalculator extends Calculator {
+    }
+    onClearButtonClick() {
+        super.onClearButtonClick()
+        this.history = ""
```
is the history array or string?

yaaHosein replied:
Array

yaaHosein replied:
I can see what I did is working even if the history is an array

yaaHosein replied:
I can see what I did is working even if the history is an array

SMH110 replied:
if it's array, then use an empty array instead of empty string

yaaHosein commented:
Just I check these when I was near the laptp
I just discommented the lines which you drew a circle around them and added to more lines of code 
I do not know why this is happened
Should I checkout to the master and git pull?

yaaHosein commented:
Array

yaaHosein commented:
I do not see anything wrong in master or feature branches
What you draw a circle around it was before and now it is on the baranch

I added line 32 and 33 in the new featur branch

yaaHosein committed:
using an empty array as a value for this.history inside clearing method

SMH110 commented on DevelopedCalculator.js:
```diff
@@ -29,7 +29,7 @@ class DevelopedCalculator extends Calculator {
     onClearButtonClick() {
         super.onClearButtonClick()
-        this.history = ""
+        this.history = [""]
```
` [""]` is this empty array?

yaaHosein replied:
I have pushed the correct one before reading this comment

yaaHosein committed:
correcting [] inside clearing method

SMH110 commented on DevelopedCalculator.js:
```diff
@@ -19,17 +19,19 @@ class DevelopedCalculator extends Calculator {
+    onClearButtonClick() {
+        super.onClearButtonClick()
+        this.history = [""]
+        this.historyElement.textContent = this.history
```
what will happen if you call `this.gettingHistoryArray();` instad of `this.historyElement.textContent = this.history`

yaaHosein committed:
calling  this.gettingHistoryArray(); inside clearing button

SMH110 commented:
Have you added a comment to your changes? will you remember the role of "onClearButtonClick" in the future?

yaaHosein committed:
adding some comments to clearing button

SMH110 commented:
you can complete this pr
